.github/workflows/ci.yml

```diff
@@ -208,37 +208,29 @@ jobs:
           distribution: adopt
           java-version: 11
 
-      - name: Local publish of artifacts
+      - name: Set up gradle cache
         uses: gradle/gradle-build-action@v2
-        with:
-          # javadoc task fails sporadically fetching https://docs.oracle.com/javase/8/docs/api/
-          arguments: publishToMavenLocal -x javadoc
+
+      - name: Local publish of artifacts
+        # javadoc task fails sporadically fetching https://docs.oracle.com/javase/8/docs/api/
+        run: ./gradlew publishToMavenLocal -x javadoc
 
       - name: Local publish of gradle plugins
-        uses: gradle/gradle-build-action@v2
-        with:
-          # javadoc task fails sporadically fetching https://docs.oracle.com/javase/8/docs/api/
-          arguments: publishToMavenLocal -x javadoc
-          build-root-directory: gradle-plugins
+        # javadoc task fails sporadically fetching https://docs.oracle.com/javase/8/docs/api/
+        run: ./gradlew publishToMavenLocal -x javadoc
+        working-directory: gradle-plugins
 
       - name: Build distro
-        uses: gradle/gradle-build-action@v2
-        with:
-          arguments: build --init-script ../../.github/scripts/local.init.gradle.kts
-          build-root-directory: examples/distro
+        run: ./gradlew build --init-script ../../.github/scripts/local.init.gradle.kts
+        working-directory: examples/distro
 
       - name: Build extension
-        uses: gradle/gradle-build-action@v2
-        with:
-          arguments: build --init-script ../../.github/scripts/local.init.gradle.kts
-          build-root-directory: examples/extension
+        run: ./gradlew build --init-script ../../.github/scripts/local.init.gradle.kts
+        working-directory: examples/extension
 
       - name: Run muzzle check against extension
-        uses: gradle/gradle-build-action@v2
-        with:
-          cache-read-only: true
```

Comment:
I realized that `cache-read-only` can't be applied separately, another good reason to move to a single `gradle/gradle-build-action@v2` step